schema/dpns-contract-documents.json

```diff
@@ -0,0 +1,130 @@
+      },
+      "preorderSalt": {
+        "type": "string",
+        "contentEncoding": "binary",
+        "minLength": 32,
+        "maxLength": 32,
+        "description": "Domain pre-order salt. 32 random bytes",
+        "$comment": "Double sha-256 of concatenated salt is used for pre-ordering document."
+      },
+      "records": {
+        "type": "object",
+        "properties": {
+          "dashUniqueIdentityId": {
+            "type": "string",
+            "minLength": 42,
+            "maxLength": 44,
+            "pattern": "^[123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz]+$",
+            "description": "Identity ID string should be used to create a primary name for Identity",
+            "$comment": "base58 string"
+          },
+          "dashAliasIdentityId": {
+            "type": "string",
+            "minLength": 42,
+            "maxLength": 44,
+            "pattern": "^[123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz]+$",
+            "description": "Identity ID string should be used to create multiple alias names for Identity",
+            "$comment": "base58 string"
+          }
+        },
+        "$comment": "Constraint with max and min properties ensure that only one identity record is used either `dashUniqueIdentityId` or `dashAliasIdentityId`",
+        "minProperties": 1,
+        "maxProperties": 1,
+        "additionalProperties": false
+      },
+      "subdomainRules": {
+        "type": "object",
+        "properties": {
+          "allowSubdomains": {
+            "type": "boolean",
+            "const": false,
+            "description": "This option defines who can create subdomains. Only domain owner is allowed at the moment"
+          }
+        },
+        "description": "Subdomain rules allows to define rules for subdomains",
+        "additionalProperties": false,
+        "required": ["allowSubdomains"]
+      }
+    },
+    "required": [
+      "label",
+      "normalizedLabel",
+      "normalizedParentDomainName",
+      "preorderSalt",
+      "records",
+      "subdomainRules"
+    ],
+    "additionalProperties": false,
+    "$comment": "normalizedLabel + '.' + normalizedParentDomain must be < 254 chars length"
+  },
+  "preorder": {
+    "indices": [
+      {
+        "properties": [
+          {
+            "saltedDomainHash": "asc"
+          }
+        ],
+        "unique": true
+      }
+    ],
+    "properties": {
+      "saltedDomainHash": {
+        "type": "string",
+        "contentEncoding": "binary",
+        "minLength": 32,
+        "maxLength": 32,
+        "description": "Double sha-256 of concatenated salt, `normalizedLabel` and `normalizedParentDomainName`. Salt should be 32 random bytes"
+      }
+    },
+    "required": [
+      "saltedDomainHash"
+    ],
+    "additionalProperties": false
+  }
```

Comment:
Let's put here the paragrah from dip that describes that you need to create a preorder and then domain and why